<?xml version="1.0"?>
<launch>
  <arg name="run_rqt_reconfigure" default="true" />

  <!-- Example launch file for the PhaseSpace ImpulseX2E motion-tracker
       calibration visualization.
       This assumes that you have already started the PhaseSpace OWL
       server and the ROS phasespace node(s). You might also need to
       check the PhaseSpace alignment transform and publish that.
       Also, set the colormap_scale to a distance that is typical
       for the calibration errors in your setup; e.g. value=0.01
       results in full-scale blue/red for distance errors of 0.01m
       or 1 centimeter.
  -->


  <node pkg="tams_phasespace" name="workspace_visualizer" type="workspace_visualizer" output="screen">
    <param name="debug_level" type="int" value="2" />
    <param name="parent_frame" type="string" value="world" />
    <param name="marker_radius" type="double" value="0.05" />
    <param name="colormap_scale" type="double" value="0.01" />
    <param name="update_rate" type="double" value="0.2" />
    <param name="enabled_cameras" type="string" value="0,1,2,3,4,5,6,7" />

    <param name="workspace_xmin" type="double" value="-1" />
    <param name="workspace_xmax" type="double" value="+1" />
    <param name="workspace_ymin" type="double" value="-1" />
    <param name="workspace_ymax" type="double" value="+1" />
    <param name="workspace_zmin" type="double" value=" 0" />
    <param name="workspace_zmax" type="double" value="+2" />

    <param name="workspace_dx" type="double" value="0.1" />
    <param name="workspace_dy" type="double" value="0.1" />
    <param name="workspace_dz" type="double" value="0.1" />

    <remap from="/workspace_visualizer/owl_cameras" to="/phasespace_ros/cameras" />
  </node>


  <group if="$(arg run_rqt_reconfigure)">
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" />
  </group>


  <!--
  <node pkg="tf" type="static_transform_publisher" name="world_broadcaster" 
        args="1 0 0 0 0 0 world phasespace 100" />
  -->

<!--  <arg unless="$(arg config)" name="command_args" value="" />
  <arg     if="$(arg config)" name="command_args" value="-d $(find taser_moveit)/launch/moveit.rviz" />
-->


  <node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen"
      args="$(find tams_phasespace)/launch/workspace.rviz" />


</launch>
